src/python/grpcio/grpc/_cython/_cygrpc/aio/call.pyx.pxi

```diff
@@ -16,149 +16,127 @@ cimport cpython
-        grpc_completion_queue_destroy(self._cq)
+        self._grpc_call_wrapper = GrpcCallWrapper()
 
     def __repr__(self):
         class_name = self.__class__.__name__
         id_ = id(self)
         return f"<{class_name} {id_}>"
 
-    @staticmethod
-    cdef void functor_run(grpc_experimental_completion_queue_functor* functor, int success) with gil:
-        pass
-
-    @staticmethod
-    cdef void watcher_call_functor_run(grpc_experimental_completion_queue_functor* functor, int success) with gil:
-        call = <_AioCall>(<CallbackContext *>functor).waiter
-
-        if not call._waiter_call.done():
-            if success == 0:
-                call._waiter_call.set_exception(Exception("Some error occurred"))
-            else:
-                call._waiter_call.set_result(None)
-
-    async def unary_unary(self, bytes method, bytes request, object timeout, AioCancelStatus cancel_status):
-        cdef grpc_call * call
+    cdef grpc_call* _create_grpc_call(self,
+                                      object timeout,
+                                      bytes method) except *:
+        """Creates the corresponding Core object for this RPC.
+
+        For unary calls, the grpc_call lives shortly and can be destroied after
+        invoke start_batch. However, if either side is streaming, the grpc_call
+        life span will be longer than one function. So, it would better save it
+        as an instance variable than a stack variable, which reflects its
+        nature in Core.
+        """
         cdef grpc_slice method_slice
-        cdef grpc_op * ops
-
-        cdef Operation initial_metadata_operation
-        cdef Operation send_message_operation
-        cdef Operation send_close_from_client_operation
-        cdef Operation receive_initial_metadata_operation
-        cdef Operation receive_message_operation
-        cdef Operation receive_status_on_client_operation
-
-        cdef grpc_call_error call_status
         cdef gpr_timespec deadline = _timespec_from_time(timeout)
-        cdef char *c_details = NULL
 
         method_slice = grpc_slice_from_copied_buffer(
             <const char *> method,
             <size_t> len(method)
         )
-
-        call = grpc_channel_create_call(
+        self._grpc_call_wrapper.call = grpc_channel_create_call(
             self._channel.channel,
             NULL,
-            0,
-            self._cq,
+            _EMPTY_MASK,
+            self._channel.cq.c_ptr(),
             method_slice,
             NULL,
             deadline,
             NULL
         )
-
         grpc_slice_unref(method_slice)
 
-        ops = <grpc_op *>gpr_malloc(sizeof(grpc_op) * _OP_ARRAY_LENGTH)
+    cdef void _destroy_grpc_call(self):
+        """Destroys the corresponding Core object for this RPC."""
+        grpc_call_unref(self._grpc_call_wrapper.call)
+
+    async def unary_unary(self, bytes method, bytes request, object timeout, AioCancelStatus cancel_status):
```

Comment:
FYI. I wanted to reduce this lengthy function into multiple reuse-able parts. So far, I refactored in from 110 to 73 LOC. Suggestions welcomed :)